Config rework

ImogenBits opened the pull request:
reworking the config parsing and CLI

Benezivas commented:
Thank you for the PR. Could you please give a small overview of the conceptional changes that this PR brings?

ImogenBits commented:
Since this is a pretty extensive PR I was planning to talk it through in our meeting later  today instead of trying to explain it through text, I should probably have written that down :D

Benezivas commented:
Thank you for the changes, there is a small bug that I noticed when running the script with the `--display=ui` option: 

```
[...]
Algobattle version 4.0.0
Battle Type: <class 'algobattle.battle_wrappers.iterated.Iterated'>
╔═══════════════╦═══════════════╦═══════╦═══════╦═══════╦═══════╦═══════╦═══════╗
║      GEN      ║      SOL      ║     1 ║     2 ║     3 ║     4 ║     5 ║   AVG ║
╠═══════════════╬═══════════════╬═══════╬═══════╬═══════╬═══════╬═══════╬═══════╣
║ c4subgraphiso ║ c4subgraphiso ║ 50000 ║       ║       ║       ║       ║ 50000 ║
╚═══════════════╩═══════════════╩═══════╩═══════╩═══════╩═══════╩═══════╩═══════╝
[...]
```
The name of the used Battlewrapper is not correctly resolved.
Is it also intended that the names of the generating and solving teams are substituted by the problem name?  
I would find it to be more reasonable to state the name of the problem seperately.

ImogenBits commented:
fixed the battle wrapper name issue.

the table does show the names of the teams, but i set it so the default team name is the name of the folder the generator and solver is in. but you're right, that usually is the name of the problem not of the team. changed it so it's just "team 0".

Benezivas commented:
Thank you, it now works as expected.
